src/libImaging/TiffDecode.c

```diff
@@ -384,29 +435,31 @@ int ImagingLibTiffDecode(Imaging im, ImagingCodecState state, UINT8* buffer, Py_
         TRACE(("TIFFTileSize: %d\n", state->bytes));
 
         for (y = state->yoff; y < state->ysize; y += tile_length) {
-            for (x = state->xoff; x < state->xsize; x += tile_width) {
-                if (ReadTile(tiff, x, y, (UINT32*) state->buffer) == -1) {
-                    TRACE(("Decode Error, Tile at %dx%d\n", x, y));
-                    state->errcode = IMAGING_CODEC_BROKEN;
-                    TIFFClose(tiff);
-                    return -1;
-                }
-
-                TRACE(("Read tile at %dx%d; \n\n", x, y));
-
-                current_tile_width = min(tile_width, state->xsize - x);
-
-                // iterate over each line in the tile and stuff data into image
-                for (tile_y = 0; tile_y < min(tile_length, state->ysize - y); tile_y++) {
-                    TRACE(("Writing tile data at %dx%d using tile_width: %d; \n", tile_y + y, x, current_tile_width));
-
-                    // UINT8 * bbb = state->buffer + tile_y * row_byte_size;
-                    // TRACE(("chars: %x%x%x%x\n", ((UINT8 *)bbb)[0], ((UINT8 *)bbb)[1], ((UINT8 *)bbb)[2], ((UINT8 *)bbb)[3]));
-
-                    state->shuffle((UINT8*) im->image[tile_y + y] + x * im->pixelsize,
-                       state->buffer + tile_y * row_byte_size,
-                       current_tile_width
-                    );
+            for (plane = 0; plane < planes; plane++) {
+                for (x = state->xoff; x < state->xsize; x += tile_width) {
+                    if (ReadTile(tiff, x, y, plane, (UINT32*) state->buffer) == -1) {
+                        TRACE(("Decode Error, Tile at %dx%d\n", x, y));
+                        state->errcode = IMAGING_CODEC_BROKEN;
+                        TIFFClose(tiff);
+                        return -1;
+                    }
+
+                    TRACE(("Read tile at %dx%d; \n\n", x, y));
+
+                    current_tile_width = min(tile_width, state->xsize - x);
+
+                    // iterate over each line in the tile and stuff data into image
+                    for (tile_y = 0; tile_y < min(tile_length, state->ysize - y); tile_y++) {
+                        TRACE(("Writing tile data at %dx%d using tile_width: %d; \n", tile_y + y, x, current_tile_width));
+
+                        // UINT8 * bbb = state->buffer + tile_y * row_byte_size;
+                        // TRACE(("chars: %x%x%x%x\n", ((UINT8 *)bbb)[0], ((UINT8 *)bbb)[1], ((UINT8 *)bbb)[2], ((UINT8 *)bbb)[3]));
+
+                        ((*unpacker)[plane])((UINT8*) im->image[tile_y + y] + x * im->pixelsize,
```

Comment:
I'm having any luck yet in coming up with a fix, but the `unpacker` pointer here is a starting point for why the macOS tests are failing.